vir/defs/high/ast/type_decl.rs

```diff
@@ -153,4 +155,5 @@ pub struct Unsupported {
 #[display(fmt = "{}", name)]
 pub struct Trusted {
     pub name: String,
+    pub fields: Vec<FieldDecl>,
```

Comment:
The trusted types should be opaque and encoded as bodyless predicates.
```suggestion
```

Reply:
I see - didn't realize I can completely remove them. The only downside of this is that I can't use `to_procedure_snapshot` on the lifetimes of trusted types - not quite sure why.

Reply:
What error do you get?

Reply:
`fn current_snapshot_variable_version` fails to unwrap the snapshot state of the current variables  - seems as if its not yet initialized. But I just realized I shouldn't use `lifetime_const_into_variable` in `owned/encoder.rs` to be more consistent.